src/spec-lang/tc/typecheck.ts

```diff
@@ -435,64 +409,120 @@ function sortContracts(contracts: ContractDefinition[]): ContractDefinition[] {
         }
     }
 
-    return topoSort(contracts, order);
+    return order.length > 0 ? topoSort(contracts, order) : contracts;
 }
 
-export function tcUnits(units: SourceUnit[], annotMap: AnnotationMap, typeEnv: TypeEnv): void {
-    let contracts: ContractDefinition[] = [];
-    const ctx: STypingCtx = [units];
+/**
+ * Get the typing context corresponding to `target` (and compiler version `version`).
+ * Note this code should work for both 0.4.x and >0.4.x variable resolution.
+ */
+export function getTypeCtx(target: AnnotationTarget, version: string): STypingCtx {
```

Comment:
I don't think we need this function, since in `lookup{VarDef,TypeDef, FunDef}` we use `resolveAny()` on the first AST node we encounter down the typing context. So we don't need the full solc-typed-ast part of the typing context - just the scribble part.

Reply:
Fixed in 5d5c2cd.